### PR TITLE
[test] Add constraint test for SR-8075.

### PR DESCRIPTION
**Pull request by huonw**

### CODE REVIEW

**Comment by huonw** [comment]
@swift-ci please smoke test

File: test/Constraints/generics.swift [hunk 1]
```diff
@@ -578,3 +578,17 @@ func rdar39616039() {
   var c = foo(default: 42.0, ["foo": Float(0)])
   c += 1 // ok
 }
+
+// https://bugs.swift.org/browse/SR-8075
+
+func sr8075() {
```
Comment (slavapestov) [inline test/Constraints/generics.swift]: Please use two-space indent for consistency with the rest of the file. Also these days we generally put random crasher test cases into `validation-test/compiler_crashers_2_fixed/`

Comment (huonw) [reply test/Constraints/generics.swift]: This wasn't a crasher.

**Comment by huonw** [comment]
@swift-ci please smoke test

**Pull request merged**